### ISSUE TITLE
Ubuntu 1404 waiting on authentication issue

### ISSUE THREAD

**Issue by jviyer** [Microsoft/AzureStorageExplorer#1046]
**Storage Exlporer 1.6.1**: 
**OS: Linux - Ubuntu 14.04**: 
**Architecture x86/64bit**: 
*This is first time installation & use **: 

#### Steps to Reproduce: ####
1. 
I installed the required Ubuntu 14.04 dependencies.
.NET runtime is as follows:
Host (useful for support):
  Version: 2.1.6
  Commit:  3f4f8eebd8
.NET Core SDKs installed:
  No SDKs were found.
.NET Core runtimes installed:
  Microsoft.NETCore.App 2.1.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

2. 
Started the Storage Explorer.
It launched the window to "Connect to Azure Storage".
Radio Button:"Add an Azure Account"
Azure environment: "Azure" drop down.
I clicked on "Sign in.."
It popped up the Sign in window. Entered my account email and then "next". It gave the password window. Entered it.
Then it entered the "waiting for authentication" loop which ended after a while but that window exited with no message about successful login or anything. Just exited, that's it. 
But then the storage explorer window does not show any account. When I go to manage accounts, my account did not show up.

**Comment by jviyer** [comment]
any help, please! ?!

**Comment by jinglouMSFT** [comment]
I believe this is specific to your machine.  Please feel free to reach out to our support team. They can help you better diagnose the issue.  Here the instruction to opening a support request:  https://docs.microsoft.com/en-us/azure/azure-supportability/how-to-create-azure-support-request

**Comment by diegodgs** [comment]
I have the same problem in ubuntu 18.04.

**Comment by maetthu** [comment]
@jinglouMSFT The issue of not being able to login from Linux has been reported several times already, e.g. in #377 or #434. This does not appear to be a specific problem of just one user - I can reproduce this as well, and judging from this and the other linked issues, it seems to be more widespread than just a couple of accounts.

**Comment by ziXet** [comment]
Same here. 16.04 after updating Ubuntu...

**Comment by cluxter** [comment]
Didn't work 2 months ago with ArchLinux on one machine (so with the latest version of Azure Storage Explorer), still not working today with ArchLinux on another machine. Here is what the console says:

![image](https://user-images.githubusercontent.com/6885295/52426987-60c41100-2aff-11e9-9897-e68ddb8f0bcc.png)

And the screen says "Waiting for authentification..." with a spinning wheel.

**Comment by Ludogo** [comment]
Same issue on ubuntu 18.04 with a fresh install. 
Is there any solution to use this tool ?

**Comment by jriouxrandstadca** [comment]
Same issue here on debian 9 with a fresh install.

**Comment by jinglouMSFT** [comment]
We recently discovered a breaking change in the .NETCore CLI that was shipped in the latest .NETCore. This change indirectly impacts the sign-in experience in Storage Explorer.  We are working with the authentication lib team to incorporate the change in the lib.  At the meantime, as a workaround, if you have the latest .NETCore (likely version 2.2.3) installed, please revert back to the lowest version of .NETCore 2.0.x available for your OS.  We've verified with .NETCore 2.0.3 on Ubuntu. 

If you don't have libgnome-keyring-common and libgnome-keyring0 installed, please install those as well.

**Comment by MRayermannMSFT** [comment]
As an alternative to @jinglouMSFT 's workaround, you can also manually patch Storage Explorer to fix the affected file. To do so you can perform the following steps:
1. Download version 1.5.43 of StreamJsonRpc [from nuget](https://www.nuget.org/packages/StreamJsonRpc/1.5.43). Look for the "Download package" link on the right hand side of the page.
2. After downloading the package, change its file extension from `.nupkg` to `.zip`
3. Unzip the package
4. Go to `streamjsonrpc.1.5.43/lib/netstandard1.1/`
5. Copy `StreamJsonRpc.dll` to the following locations inside the Storage Explorer folder:
    1. `StorageExplorer/resources/app/ServiceHub/Services/Microsoft.Developer.IdentityService/`
    2. `StorageExplorer/resources/app/ServiceHub/Hosts/ServiceHub.Host.Core.CLR.x64/`

If you do that, you shouldn't need to downgrade your .NET Core version. We'll be updating out Linux docs later today with updated Linux install instructions.

**Comment by jriouxrandstadca** [comment]
Perfect thank you. I will try that later today 👍

**Comment by MRayermannMSFT** [comment]
Here are the updated docs: https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-troubleshooting?tabs=1804#linux-dependencies

**Comment by marcorodrigues19** [comment]
I was having the same issue on Linux 18.04 

@MRayermannMSFT It worked for me, thx for the help.

**Comment by MRayermannMSFT** [comment]
Sounds like everyone is unblocked. Glad we could help. New auth library bits have been merged into master for 1.8.0 so future versions of Storage Explorer will not need the patch. Thank you for the patience. 😃
